rfc/plugins.md

```diff
@@ -0,0 +1,206 @@
+# Plugins for ember-cli-deploy
+
+This document describes the API related to plugins in the next
+version of ember-cli-deploy.
+
+### Overview of plugins
+
+A plugin is an ember-cli addon that hooks into the ember-cli-deploy
+pipeline in order to add functionality to a deploy. Example addon
+functionality would be uploading assets to S3, writing the index.html to
+Redis, or notifying a Slack channel a deploy has completed.
+
+In general, OSS plugins should focus on doing a specific task. Most
+Ember developers with common deployment targets will compose multiple
+plugins to fine tune their deployment process. Developers with very
+custom needs might create a single private plugin that implements all
+aspects of their deployment process within the structure provided by
+ember-cli-deploy.
+
+Because plugins are implemented via addons, they may be included via
+node_module dependencies or via in-repo-addons.
+
+### Identifying plugins
+
+Plugins are ember-cli addons. They will also have the keyword
+`ember-cli-deploy-plugin`. An example `package.json` for a plugin:
+
+```
+{
+  "name": "ember-cli-deploy-example-plugin",
+  "version": "0.4.0",
+  // ...
+  "devDependencies": {
+    "ember-cli": "0.2.0",
+    "ember-cli-deploy": "0.4.1"
+    // ...
+  },
+  "keywords": [
+    "ember-addon",
+    "ember-cli-deploy-plugin"
```

Comment:
@achambers I'm pretty sure it is accessible at `addon.pkg.keyworks`. That said, your suggestion of looking for that method seems reasonable, too. We don't get easy npm filtering without the keyword though...